packages/muelu/src/Graph/MatrixTransformation/MueLu_CoalesceDropFactory_def.hpp

```diff
@@ -833,13 +833,19 @@ namespace MueLu {
               }
 
               LO nnz = indices.size();
+              bool haveAddedToDiag = false;
               for (LO colID = 0; colID < nnz; colID++) {
                 const LO col = indices[colID];
 
                 if (row != col) {
                   localLaplDiagData[row] += STS::one() / MueLu::Utilities<real_type,LO,GO,NO>::Distance2(coordData, row, col);
+                  haveAddedToDiag = true;
```

Comment:
@cgcgcg I find this variable name a little confusing. To me, its name indicates that it will be `true` when something has been added to the diagonal. Yet, here it's set to `true` if `row!=col`, i.e. if something has been added to the off-diagonal.

Reply:
Things are being lumped from off-diagonal to diagonal..